### ISSUE TITLE
Adapting a yolo model to real life scenario

### ISSUE THREAD

**Issue by rezktr** [ultralytics/ultralytics#22476]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I trained a YOLO model to recognize playing cards. To do this, I took photos of playing cards, converted them into PNGs, and generated 50,000 synthetic and clean images from these PNGs. Then, I created another 50,000 images containing camera distortions where the objects appeared distorted, and I trained my model with these images. The training metrics were good, but it wasn't very effective in a real-world scenario. To overcome this, do I need to capture images from the specific cameras that will be used for scanning and retrain a new model with these images? What can I do to speed up the labeling process? I am open to your suggestions.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rezktr, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you the fastest support possible—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a custom training ❓ Question, please share more details so we can provide targeted guidance:
- What model and version are you using (e.g., YOLO11n/s/m/l/x)?
- Training command and key settings (epochs, imgsz, augmentation settings)
- Sample images that represent the real-world camera conditions you’re targeting
- A few representative failure cases (images + predictions) and training/validation logs
- Confirmation that you’ve reviewed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the detailed context—this looks like a classic synthetic-to-real domain gap. Yes, capture a small, diverse set of real frames from each deployment camera and fine-tune your existing weights; also consider calibrating and undistorting inputs rather than training on heavily warped renders, which often transfers better to production; see our camera calibration guide: [2025 Vision AI Camera Calibration](https://www.ultralytics.com/blog/a-guide-to-camera-calibration-for-computer-vision-in-2025).  

To speed labeling, auto-label with your current detector and then correct the outputs:

```bash
yolo predict model=path/to/your_detector.pt source=real_images/ save_txt=True save_conf=True save_crop=True
```

Then fine-tune on the corrected labels:

```bash
yolo train model=path/to/your_detector.pt data=cards.yaml imgsz=640 epochs=20 cache=True
```

If a single model over 52 classes struggles, consider a two-stage setup (detect “card” as one class, then classify rank/suit on crops), which tends to generalize better across decks; see approaches in [Using a Vision AI model to recognize playing cards](https://www.ultralytics.com/blog/using-a-vision-ai-model-to-recognize-playing-cards). For robust augmentations on real images, integrate Albumentations as outlined here: [Ultralytics YOLO Data Augmentation with Albumentations](https://www.ultralytics.com/blog/using-albumentations-augmentations-to-diversify-your-data). Finally, measure progress on a real hold-out set per our [Model Testing guide](https://docs.ultralytics.com/guides/model-testing/).

**Comment by rezktr** [comment]
@glenn-jocher thanks a lot for your help. I also have one more question, in my scenario my camera footage has 1920*1080 res however the classes that appear in the footage takes up 10*21 px space is this an issue or yolov8-obb can detect this small objects when nicely fine-tuned?

**Comment by glenn-jocher** [comment]
Great question—10×21 px at 1920×1080 becomes ~3×7 px if you resize to 640, which is below the model’s effective stride, so recall will be poor regardless of fine‑tuning; make the targets larger at the network input by using higher `imgsz` or sliced/tiling inference.

```bash
# Inference at native scale (keeps small objects resolvable)
yolo predict model=path/to/weights.pt source=video.mp4 imgsz=1920
```

OBB helps with rotation but doesn’t change the minimum pixel-size requirement; if angle matters use YOLO11 OBB per the [YOLO11 OBB Detection Guide](https://www.ultralytics.com/blog/how-to-use-ultralytics-yolo11-for-obb-object-detection), and for tiny targets use tiled inference via our [SAHI Tiled Inference guide](https://docs.ultralytics.com/guides/) and the “image tiling for small objects” tips in [Model Evaluation Insights](https://docs.ultralytics.com/guides/model-evaluation-insights/).

**Comment by rezktr** [comment]
@glenn-jocher thanks a lot, i didnt know yolo11-obb existed, If I use the original image size, will 10*21 px be enough?

**Comment by glenn-jocher** [comment]
Short answer: it’s borderline—objects ~10×21 px at 1920×1080 are at the edge of reliable detection; OBB won’t change the minimum size requirement, it just models rotation. Make the cards effectively larger at the network input (tile/zoom or raise `imgsz` if GPU allows) so each card is ≥20–30 px on its shortest side, and train/val at that same scale. For setup details see the [Oriented Bounding Boxes docs](https://docs.ultralytics.com/tasks/obb/).

**Comment by rezktr** [comment]
Thanks a lot

**Comment by glenn-jocher** [comment]
You're welcome—if you want to speed up annotation on your real-camera frames, try the auto-labeling workflows in [Ultralytics HUB](https://docs.ultralytics.com/hub/); if anything unexpected comes up, please share a minimal sample and the output of `yolo checks` for review.
